python/ray/tune/trial_runner.py

```diff
@@ -271,6 +286,17 @@ def checkpoint(self, force=False):
             json.dump(runner_state, f, indent=2, cls=_TuneFunctionEncoder)
 
         os.rename(tmp_file_name, self.checkpoint_file)
+
+        try:
+            ckpt_tmp_file_name = os.path.join(self._local_checkpoint_dir,
```

Comment:
Do you think it makes sense to just include the search alg state in `checkpoint_file` (ie put it in `runner_state`) since it looks like checkpointing/restoring the rest of the runner state + search alg state are always done together?

Reply:
If you keep it as is you probably also want to add a few asserts to tests in `test_trial_runner.py` (eg `testCheckpointOverwrite`, `testCheckpointWithFunction`) to make sure checkpoints are working correctly for the searcher.

Reply:
Yeah that was smth I was discussing with Richard, the way the current save/restore API works is it ckpts / restores from a given dir, so it would have to be overhauled to ckpt/restore to/from a section of a .json file. Seemed simpler to just add a second file in same ckpt'ing directory to work with.

As for testing, test_cluster.py seems like a good template to go from - does a ckpt and confirms file is created. Working on something here: https://github.com/hershg/ray/commit/9c747f56071957ca61730e96ab79b7a16d6008db but test is WIP